src/pipeline/common/Manifest.groovy

```diff
@@ -1,274 +1,272 @@
+def downloadManifest(String url, String target_dir){
+    fname= sh ( script:  """var=$url &&   echo \"\${var##*/}\"  """, returnStdout: true ).trim()
+    sh 'wget --retry-connrefused --waitretry=1 --read-timeout=20 --timeout=15 -t 1 ' +  "$url" + ' -P ' + "${target_dir}"
+    return target_dir +  File.separator + fname 
+}
 
 
-    // ==================================================================
-    // stash a manifest file: to save the file(s) in Jenkins server for later usage from other vmslave
-    //
-    // Inputs: 
-    // stash_name   : the "key" of the locker
-    // manifest_path: where the manifest file is located (path/file)
-    //
-    // Output:
-    // a Groovy Map, in which the "stash_name" is the key, and "stash_path" is the path (useful to find the files after unstash)
-    // ==================================================================
-    def stashManifest(String stash_name, String manifest_path){
-        file_path = manifest_path
-        // using regex to check if manifest_path is absolute path, Note: Java File.isAbsolute is restriced in Jenkins runtime
-        is_relative_path= sh (  returnStdout: true ,
-                      script: """#!/bin/bash -e
-                      if [[  \"$manifest_path\" != /* ]] || [ \"\$(basename $manifest_path)\" == $manifest_path ] ; then
-                            echo yes;
-                      fi""").trim()
-        if ( is_relative_path != "yes" )
-        {
-             // *******************
-             // NOTE: Jenkins Stash has limitation, it can't stash file either from a absolute path or file outside the Jenkins $WORKSPACE
-             // here, WORKSPACE is a build-in variable in Jenkins runtime
-             // *******************
-             // and we will have to make it flat
-             sh """#!/bin/bash -e
-                msg="[Warning]copy from $manifest_path to $WORKSPACE failed.maybe copy destination and source are the same, if so, please ignore the copy error message. "
-                cp $manifest_path $WORKSPACE || echo \$msg
-                """
-             file_path = sh ( script: "echo \$(basename $manifest_path)",  returnStdout: true ).trim()
-        }
-        stash name: stash_name, includes: file_path
-        manifest_dict = [:]
-        manifest_dict["stash_name"] = stash_name
-        manifest_dict["stash_path"] = file_path 
-        return manifest_dict
+// ==================================================================
+// stash a manifest file: to save the file(s) in Jenkins server for later usage from other vmslave
+//
+// Inputs: 
+// stash_name   : the "key" of the locker
+// manifest_path: where the manifest file is located (path/file)
+//
+// Output:
+// a Groovy Map, in which the "stash_name" is the key, and "stash_path" is the path (useful to find the files after unstash)
+// ==================================================================
+def stashManifest(String stash_name, String manifest_path){
```

Comment:
Is stashManifest different from a common stashFile?

Reply:
actually. not .